forge/csrc/passes/erase_inverse_ops.cpp

```diff
@@ -306,37 +311,44 @@ void commute_and_bypass(graphlib::Graph *graph, std::vector<graphlib::Node *> co
             // (1024,) Placing an unsqueeze clone will cause the input to be (1, 1024) which isn't wrong but also is not
             // correct. In this case we shall convert the clone to a reshape which contains the correct number of
             // unsqueezes implicitly.
-            if ((first->op_name() == "unsqueeze" or first->op_name() == "squeeze") and
-                dynamic_cast<graphlib::InputNode *>(graph->node_by_id(operand_edge.producer_node_id)))
+            if ((first->op_name() == "unsqueeze" or first->op_name() == "squeeze"))
             {
                 op->change_op_type("reshape");
-                graphlib::Shape op_shape = graphlib::Shape::create(std::vector<uint32_t>(consumer->shape().size(), 1));
-                auto input = dynamic_cast<graphlib::InputNode *>(graph->node_by_id(operand_edge.producer_node_id));
-
-                for (int i = -1; i >= -(int)input->shape().size(); i--)
+                std::vector<graphlib::Edge> clone_data_edges = graph->operand_data_edges(clone);
+                for (const graphlib::Edge &clone_data_edge : clone_data_edges)
```

Comment:
Won't this clone_data_edges be empty since clone isn't added between any two nodes in the graph untill `insert_node_on_edge` call later? So will we ever enter this for loop?

Reply:
Any thoughts on this? @ashokkumarkannan1

Reply:
@dgolubovicTT   I'm checking on this. Now I moved insert_node_on_edge before this block and now I can get the proper data edges and its not empty. If you have any docs on this commute_and_bypass, please share it with me.

Reply:
I don't have any additional docs, just reading the code...